src/App.tsx

```diff
@@ -1,223 +1,144 @@
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorTypes } from './types/ErrorTypes';
 
 export const App: React.FC = () => {
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState(ErrorTypes.None);
+  const [filterBy, setFilterBy] = useState(Filter.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [loadingList, setLoadingList] = useState<number[]>([]);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+  const getTodosFromServer = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setHasLoadingError(true);
+        setErrorMessage(ErrorTypes.LOADING);
+      }
     }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
   }, []);
 
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  let visibleTodos = todos;
+
+  switch (filterBy) {
+    case Filter.COMPLETED:
+      visibleTodos = completedTodos;
+      break;
+
+    case Filter.ACTIVE:
+      visibleTodos = activeTodos;
+      break;
+
+    default: visibleTodos = todos;
+  }
+
+  const errorInfo = (errorTitle: ErrorTypes) => {
+    setHasLoadingError(true);
+    setErrorMessage(errorTitle);
+  };
+
+  const clearErrorMessage = () => setHasLoadingError(false);
+
+  const addNewTodo = (newTodo: Todo) => setTodos(
+    (prevTodos) => [...prevTodos, newTodo],
+  );
+
+  const addTodoToLoadingList = (idToAdd: number) => setLoadingList(
+    prevIds => [...prevIds, idToAdd],
+  );
+
+  const deleteTodoOfLoadingList = (idToRemove: number) => setLoadingList(
+    prevIds => prevIds.filter(id => id !== idToRemove),
+  );
+
+  const clearLoadingList = () => setLoadingList([]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+      {errorMessage === ErrorTypes.LOADING
+        ? (
+          <>
+            <span>Error... Can not load your todos</span>
+          </>
```

Comment:
do you really need this?

Reply:
Yes, that is the idea that if the user has problems loading his todos, then the todos list will not be displayed,
but maybe you're right